sklearn/utils/validation.py

```diff
@@ -1418,6 +1419,16 @@ def check_scalar(
             f"Possible values are: {expected_include_boundaries}."
         )
 
+    if max_val is None and include_boundaries in ("right"):
+        raise ValueError(
+            "`include_boundaries`='right' with no `max_val` is inconsistent."
+        )
+
+    if min_val is None and include_boundaries in ("left"):
+        raise ValueError(
```

Comment:
The codecov failure is indeed normal. Here, you added some code but we never use it in the library and thus we don't know if it behave as expected.

Therefore, you need to add some new tests in `sklearn/utils/tests/test_validation.py`. There is a function called `test_check_scalar_invalid` where you can add the cases that you should raise the two error to confirm that we properly raise the error.